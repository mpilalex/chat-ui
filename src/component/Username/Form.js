import React from "react";
import PropTypes from "prop-types";
import Component from "./../BaseComponent";
import {Icon, Form, Button} from "semantic-ui-react";
import TextField from "../Form/TextField";
import {reduxForm, Field} from "redux-form";

class UsernameForm extends Component{
    render(){
        const {handleSubmit} = this.props;
        return <Form className={"username-form"} onSubmit={handleSubmit}>
            <Field name="username" component={TextField} />
            <Button icon><Icon name={"checkmark"} /></Button>
        </Form>
    }
}

UsernameForm.propTypes = {
    onSubmit: PropTypes.func.isRequired
};

export default reduxForm({
    form: "username"
})(UsernameForm);