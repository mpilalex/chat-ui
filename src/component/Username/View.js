import React from "react";
import PropTypes from "prop-types";
import Component from "./../BaseComponent";
import {Button, Icon} from "semantic-ui-react";


class UsernameView extends Component{
    render(){
        const {username, editMode} = this.props;
        return <div className={"username-view"}>
            {username}
            <Button icon onClick={editMode}><Icon name={"write"} /></Button>
        </div>
    }
}

UsernameView.propTypes = {
    username: PropTypes.string.isRequired,
    editMode: PropTypes.func.isRequired
};

export default UsernameView;