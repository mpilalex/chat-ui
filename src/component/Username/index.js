import React from "react";
import Component from "../BaseComponent";
import PropTypes from "prop-types";
import View from "./View";
import Form from "./Form";

class Username extends Component{

    state = {
        edit: false
    };

    componentWillMount(){
        const {username} = this.props;
        if(!username){
            this.setState({edit: true})
        }
    }

    componentWillReceiveProps(nextProps){
        if(nextProps.username && !this.props.username){
            this.setState({edit: false})
        }
    }

    editMode(){
        this.setState({edit: true})
    }

    viewMode(){
        this.setState({edit: false})
    }

    onUpdate(values){
        const {onUpdate} = this.props;
        onUpdate(values);
        this.viewMode();
    }

    render(){
        const {edit} = this.state;
        const {username} = this.props;
        if(edit){
            return <Form
                initialValues={{
                    username
                }}
                onSubmit={this.onUpdate.bind(this)}
            />
        }
        return <View
            editMode={this.editMode.bind(this)}
            username={username} />
    }
}

Username.propTypes = {
    username: PropTypes.string,
    onUpdate: PropTypes.func.isRequired
};


export default Username;