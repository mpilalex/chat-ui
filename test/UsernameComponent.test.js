import React from 'react';
import Username from '../src/component/Username';
import renderer from 'react-test-renderer';
import {Provider} from "react-redux";
import reduxStore from "./../src/store";

test('username render', () => {

    const usernameFormComponent = renderer.create(<Provider store={reduxStore}><Username onUpdate={() => {}} /></Provider>);
    const usernameFormTree = usernameFormComponent.toJSON();
    expect(usernameFormTree).toMatchSnapshot();

    const usernameViewComponent = renderer.create(<Provider store={reduxStore}><Username username={"epictetus"} onUpdate={() => {}} /></Provider>)
    const usernameViewTree = usernameViewComponent.toJSON();
    expect(usernameViewTree).toMatchSnapshot();

});
