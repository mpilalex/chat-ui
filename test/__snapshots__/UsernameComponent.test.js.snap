// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`username render 1`] = `
<form
  action={undefined}
  className="ui form username-form"
  onSubmit={[Function]}
>
  <div
    className="ui input"
  >
    <input
      disabled={undefined}
      onChange={[Function]}
      tabIndex={undefined}
      type="text"
      value=""
    />
  </div>
  <button
    className="ui icon button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
  >
    <i
      aria-hidden="true"
      className="checkmark icon"
    />
  </button>
</form>
`;

exports[`username render 2`] = `
<div
  className="username-view"
>
  epictetus
  <button
    className="ui icon button"
    disabled={undefined}
    onClick={[Function]}
    role="button"
    tabIndex={undefined}
  >
    <i
      aria-hidden="true"
      className="write icon"
    />
  </button>
</div>
`;
